CMakeLists.txt

```diff
@@ -1153,9 +1145,11 @@ if(NOT WIN32 OR ROCKSDB_INSTALL_ON_WINDOWS)
 endif()
 
 option(WITH_ALL_TESTS "Build all test, rather than a small subset" ON)
-
 if(WITH_TESTS OR WITH_BENCHMARK_TOOLS)
+  set(saved_cxx_flags "${CMAKE_CXX_FLAGS}")
+  string(REPLACE " -Werror" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```

Comment:
Is it possible to use a flag to control this behavior, similar to `DISABLE_WARNING_AS_ERROR` in Makefile build?

Reply:
we already have an option named `FAIL_ON_WARNINGS` for enabling/disabling `-Werror` . i am fine if you think we should just pass `-DFAIL_ON_WARNINGS=OFF` when compiling RocksDB with GCC-11.

or, an alternative would be upgrade the bundled gtest to a newer version. but since we have done this before, and that change was reverted by afa3518839a013173ee90599aab6f6f5ae58f5f7. guess we are not there yet.

please let me know what you think. and i'd be happy to drop this change if you think `-D FAIL_ON_WARNINGS=OFF` is good enough.

Reply:
I would suggest dropping this change and use `-DFAIL_ON_WARNINGS=OFF`.

Reply:
@riversand963 dropped and repushed.